### PR TITLE
Add unit tests for Python to Engine format conversion

### PR DESCRIPTION
**Pull request by par4m**
Implements https://github.com/cocoindex-io/cocoindex/issues/184

### CODE REVIEW

File: python/cocoindex/tests/test_convert.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    u = uuid.uuid4()
+    assert to_engine_value(u) == u.bytes
+
+def test_to_engine_value_date_time():
+    d = datetime.date(2024, 1, 1)
+    t = datetime.time(12, 30)
+    dt = datetime.datetime(2024, 1, 1, 12, 30)
+    assert to_engine_value(d) == d
+    assert to_engine_value(t) == t
+    assert to_engine_value(dt) == dt
```
Comment (badmonster0) [inline python/cocoindex/tests/test_convert.py:39]: nit: We may interleave each assignments with assert (i.e. `assert to_engine_value(d) == d` follows `d = datetime.date(2024, 1, 1)` directly). It makes define/use of each variables closer.

File: python/cocoindex/tests/test_convert.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    assert to_engine_value(basket) == [["apple", "banana"]]
+
+def test_to_engine_value_nested_struct():
+    customer = Customer(name="Alice", order=Order("O1", "item1", 10.0))
+    assert to_engine_value(customer) == ["Alice", ["O1", "item1", 10.0]]
+
+def test_to_engine_value_empty_list():
+    assert to_engine_value([]) == []
+    assert to_engine_value([[]]) == [[]]
+
```
Comment (badmonster0) [inline python/cocoindex/tests/test_convert.py:60]: Will be great if also adding a test case for tuple, including empty tuple (result should be similar to list)

**Review by badmonster0 — COMMENTED** [review]
Thank you!

File: python/cocoindex/tests/test_convert.py (hunk 2 above)
Comment (par4m) [reply python/cocoindex/tests/test_convert.py:60]: Done! Let me know if there’s anything else you'd like me to adjust. Thanks!

**Review by badmonster0 — APPROVED** [review]
Awesome! Thanks you very much!
